packages/dev/core/src/FlowGraph/Blocks/Execution/flowGraphPlayAnimationBlock.ts

```diff
@@ -0,0 +1,115 @@
+import type { FlowGraphContext } from "../../flowGraphContext";
+import type { Animatable, Animation, IAnimatable } from "../../../Animations";
+import { FlowGraphConnectionType } from "../../flowGraphConnection";
+import { FlowGraphDataConnection } from "../../flowGraphDataConnection";
+import { FlowGraphSignalConnection } from "../../flowGraphSignalConnection";
+import { FlowGraphAsyncExecutionBlock } from "../../flowGraphAsyncExecutionBlock";
+import { RichTypeAny, RichTypeNumber, RichTypeBoolean } from "../../flowGraphRichTypes";
+
+/**
+ * @experimental
+ * A block that plays an animation on an animatable object.
+ */
+export class FlowGraphPlayAnimationBlock extends FlowGraphAsyncExecutionBlock {
+    /**
+     * The target to play the animation on.
+     */
+    public readonly target: FlowGraphDataConnection<IAnimatable>;
+    /**
+     * The animation to play.
+     */
+    public readonly animation: FlowGraphDataConnection<Animation>;
+    /**
+     * The speed of the animation.
+     */
+    public readonly speed: FlowGraphDataConnection<number>;
+    /**
+     * Should the animation loop?
+     */
+    public readonly loop: FlowGraphDataConnection<boolean>;
+    /**
+     * The starting frame of the animation.
+     */
+    public readonly from: FlowGraphDataConnection<number>;
+    /**
+     * The ending frame of the animation.
+     */
+    public readonly to: FlowGraphDataConnection<number>;
+
+    /**
+     * The signal that is triggered when the animation ends.
+     */
+    public readonly onAnimationEnd: FlowGraphSignalConnection;
+
+    public constructor() {
+        super();
+
+        this.target = new FlowGraphDataConnection("target", FlowGraphConnectionType.Input, this, RichTypeAny);
+        this.animation = new FlowGraphDataConnection("animation", FlowGraphConnectionType.Input, this, RichTypeAny);

```

Comment:
No rich type for animations?

Maybe RichTypeObject?

Reply:
Would it have any advantages over RichTypeAny? I don't think we can create a "default" animation so undefined feels more appropriate to me...